packages/ptokens-assets-evm/src/ptokens-evm-provider.ts

```diff
@@ -30,43 +34,39 @@ export type MakeContractCallOptions = {
   abi: Abi
   /** The contract address. */
   contractAddress: string
+  /** Call on the interim chain */
+  onInterim?: boolean
 }
 
-// class SendObject<T> {
-//   account: `0x${string}`
-//   address: `0x${string}`
-//   abi: Abi
-//   functionName: string
-//   value: string
-//   gasPrice: string
-//   gas: string
-//   args: [] | T
-//   constructor(value: string) {
-//     this.value = value
-//   }
-//   maybeSetGasPrice(account: `0x${string}`, address:`0x${string}`, gasPrice: number) {
-//     if (gasPrice) this.gasPrice = gasPrice.toString()
-//     return this
-//   }
-//   maybeSetGasLimit(gasLimit: number) {
-//     if (gasLimit) this.gas = gasLimit.toString()
-//     return this
-//   }
-// }
-
 export class pTokensEvmProvider implements pTokensAssetProvider {
   private _publicClient: PublicClient
-  private _walletClient: WalletClient
-  private _gasPrice: number
-  private _gasLimit: number
+  private _interimClient: PublicClient
```

Comment:
See below. One provider should have one client.
If you need to access the interim chain, create a pTokensEvmProvider running on the interim chain. Simpler and easier to maintain